packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1287,20 +1286,57 @@ abstract class ResidentRunner {
   Future<void> maybeCallDevToolsUriServiceExtension() async {
     _devToolsLauncher ??= DevtoolsLauncher.instance;
     if (_devToolsLauncher.activeDevToolsServer != null) {
-
-      await Future.wait(<Future<vm_service.Isolate>>[
+      await Future.wait(<Future<void>>[
         for (final FlutterDevice device in flutterDevices)
-          waitForExtension(device.vmService, 'ext.flutter.activeDevToolsServerAddress'),
+          _callDevToolsUriExtension(device),
       ]);
+    }
+  }
+
+  Future<void> _callDevToolsUriExtension(FlutterDevice device) async {
```

Comment:
For both of these - since they are unawaited, I would make sure they don't run if the resident runner is going to get torn down immediately. Maybe add back the `_devToolsLauncher =null` in tear down, and then do an early return if it is null

Reply:
added early return to `_callDevToolsUriExtension`